src/main/java/org/sagebionetworks/bridge/dynamodb/DynamoParticipantDataDao.java

```diff
@@ -0,0 +1,130 @@
+package org.sagebionetworks.bridge.dynamodb;
+
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBMapper;
+import com.amazonaws.services.dynamodbv2.datamodeling.DynamoDBQueryExpression;
+import com.amazonaws.services.dynamodbv2.datamodeling.PaginatedQueryList;
+import com.amazonaws.services.dynamodbv2.datamodeling.QueryResultPage;
+import com.amazonaws.services.dynamodbv2.model.AttributeValue;
+import com.amazonaws.services.dynamodbv2.model.Condition;
+import com.google.common.collect.Iterables;
+import com.google.common.collect.Lists;
+import org.sagebionetworks.bridge.BridgeUtils;
+import org.sagebionetworks.bridge.dao.ParticipantDataDao;
+import org.sagebionetworks.bridge.models.ForwardCursorPagedResourceList;
+import org.sagebionetworks.bridge.models.PagedResourceList;
+import org.sagebionetworks.bridge.models.ParticipantData;
+import org.sagebionetworks.bridge.models.ResourceList;
+import org.sagebionetworks.bridge.models.reports.ReportData;
+import org.springframework.stereotype.Component;
+
+import javax.annotation.Resource;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkNotNull;
+
+@Component
+public class DynamoParticipantDataDao implements ParticipantDataDao {
+    private DynamoDBMapper mapper;
+
+    @Resource(name = "participantDataMapper")
+    final void setParticipantStateMapper(DynamoDBMapper participantStateMapper) {
+        this.mapper = participantStateMapper;
+    }
+
+    @Override
+    public ForwardCursorPagedResourceList<? extends ParticipantData> getParticipantData(String userId, String offsetKey, int pageSize) {
+        checkNotNull(userId);
+
+        DynamoParticipantData hashKey = new DynamoParticipantData();
+        hashKey.setUserId(userId);
```

Comment:
If the offsetKey exists, it's going to be the identifier and you'll want to set that in the key.

Reply:
Actually, with query, it's a bit simpler than that. We can use query.setExclusiveStartKey() and pass in the offsetKey. The offsetKey would just be the last dataId in the result set.

Reply:
EDIT: I mispoke. exclusiveStartKey only works with queryPage, and it's a complex opaque DynamoDB object that we may or may not want to expose to API callers.

It would be better to use a rangeKeyCondition. For example:
```
        Condition rangeKeyCondition = new Condition().withComparisonOperator(ComparisonOperator.GT)
                .withAttributeValueList(new AttributeValue().withS(dataId));
        query.withRangeKeyCondition("dataId", rangeKeyCondition);
```